src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -661,11 +661,14 @@ private BoundExpression BoxReceiver(BoundExpression rewrittenReceiver, NamedType
         ///        l += goo(ref l);
         /// 
         /// even though l is a local, we must access it via a temp since "goo(ref l)" may change it
-        /// on between accesses. 
+        /// on between accesses.
+        ///
+        /// Note: In `this.x++`, `this` cannot change between reads. But in `(this, ...) == (..., this.Mutate())` it can.
```

Comment:
You generally need more context to be sure that `this` does not change between reads. - At least you need to be sure that the instance is not on the heap (field in a class, array element or captured). 
` = false`  seems aggressive.

Reply:
The reason for `= false` default is that is the current behavior (before this PR). To my knowledge, only deconstruction and tuple-equality need the new behavior.
Would you have an example or scenario to clarify your concern?